api/core/storage/mongo/partials/mongo_organizations.py

```diff
@@ -404,6 +404,23 @@ async def feedback_slack_hook_for_tenant(self, tenant_uid: int) -> str | None:
             raise ObjectNotFoundException("Organization not found", code="organization_not_found")
         return doc.get("feedback_slack_hook")
 
+    @override
+    async def get_organization_by_slack_channel_id(self, slack_channel_id: str) -> TenantData | None:
+        cursor = self._collection.find({"slack_channel_id": slack_channel_id})
+        docs: list[OrganizationDocument] = []
+        async for doc in cursor:
+            docs.append(OrganizationDocument.model_validate(doc))  # noqa: PERF401
+
+        if not docs:
+            return None
+
+        if len(docs) > 1:
```

Comment:
Let's just make the index unique

Reply:
I now use _get_public_org in https://github.com/WorkflowAI/WorkflowAI/pull/176/commits/63e63a9ee9b79e05667a321b73e4b8ff7632dd33